File: client/types/payload/request/set.password.request.ts
import { z } from "zod/v3";

export const SetPasswordRequest = z
  .object({
    password: z
      .string({ required_error: "Password cannot be blank" })
      .min(1, { message: "Password cannot be blank" })
      .min(6, {
        message: "Password is too short. Please enter at least 6 characters",
      })
      .max(100, {
        message:
          "Password is too long. Please enter no more than 100 characters",
      }),
    confirmPassword: z.string(),
  })
  .refine((data) => data.password === data.confirmPassword, {
    message: "Password does not match",
    path: ["confirmPassword"],
  })
  .transform(({ confirmPassword, ...data }) => data);

export type SetPasswordRequest = z.infer<typeof SetPasswordRequest>;

File: backend/src/main/java/com/lerneon/backend/models/payload/request/SetPasswordRequest.java
package com.lerneon.backend.models.payload.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class SetPasswordRequest {
    @NotBlank(message = "Password cannot be blank")
    @Size(min = 6, message = "Password is too short. Please enter at least 6 characters")
    @Size(max = 100, message = "Password is too long. Please enter no more than 100 characters")
    private String password;
}
